### ISSUE TITLE
Consolidation is blocked by pods that have terminationGracePeriodSeconds

### ISSUE THREAD

**Issue by thjwhite** [aws/karpenter#4809]
### Description

**Observed Behavior**:
We have an EKS cluster where we observed very slow consolidation. In the logs of karpenter we see frequent gaps of 5+ minutes between when consolidation cordons a node and when it deletes it. There is an application running on these nodes which require a terminationGracePeriodSeconds of 5min. The application in question can have hundreds of pods split across dozens of nodes.  When I correlate the logs on each node, I see that in the cases where this gap exists - this app is running on that node. This seems like the correct behavior, since the application needs to gracefully shutdown, however,  karpenter does not take any other consolidation actions across the cluster in that time while it is waiting on the graceful shutdown of a single pod.

>  2023-10-11 18:28:03 | (no unique labels) | 2023-10-12T01:28:03.837Z	INFO	controller.termination	deleted node	{"commit": "322822a", "node": "ip-10-43-138-24.ec2.internal"}
-- | -- | --
  |   | 2023-10-11 18:23:00 | (no unique labels) | 2023-10-12T01:23:00.846Z	INFO	controller.termination	cordoned node	{"commit": "322822a", "node": "ip-10-43-138-24.ec2.internal"}

On cases where the gap is small, that application is not running on those nodes:

> 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.607Z	INFO	controller.termination	deleted node	{"commit": "322822a", "node": "ip-10-43-191-64.ec2.internal"}
-- | -- | --
  |   | 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.606Z	INFO	controller.termination	deleted node	{"commit": "322822a", "node": "ip-10-43-191-76.ec2.internal"}
  |   | 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.606Z	INFO	controller.termination	deleted node	{"commit": "322822a", "node": "ip-10-43-140-203.ec2.internal"}
  |   | 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.035Z	INFO	controller.termination	cordoned node	{"commit": "322822a", "node": "ip-10-43-140-203.ec2.internal"}
  |   | 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.022Z	INFO	controller.termination	cordoned node	{"commit": "322822a", "node": "ip-10-43-191-76.ec2.internal"}
  |   | 2023-10-11 17:52:53 | (no unique labels) | 2023-10-12T00:52:53.016Z	INFO	controller.termination	cordoned node	{"commit": "322822a", "node": "ip-10-43-191-64.ec2.internal"}

We also observed in cases where the 5 min gap is present, it only ever deprovisions one node at a time. When there is not a gap, multiple nodes are deprovisioned within a short amount of time. 

Additionally when visually looking at the output of `kubectl describe node` we see many nodes with small utilization and overall binpacking efficiency is reduced.

**Expected Behavior**:
Consolidation deprovisions nodes in a timely manner. 

**Reproduction Steps** (Please include YAML):

**Versions**:
- Chart Version: 0.31.0
- Kubernetes Version (`kubectl version`):  

> Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.1", GitCommit:"4c9411232e10168d7b050c49a1b59f6df9d7ea4b", GitTreeState:"clean", BuildDate:"2023-04-14T13:14:41Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.12-eks-2d98532", GitCommit:"0aa16cf4fac4da27b9e9e9ba570b990867f6a3d8", GitTreeState:"clean", BuildDate:"2023-07-28T16:52:04Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
This feels like a duplicate of https://github.com/aws/karpenter-core/issues/370 since we're capturing the conversation around improved performance for deprovisioning (which includes support for parallel deprovisioning). We're actually planning on attacking the problem that you are describing here first with this PR: https://github.com/aws/karpenter-core/pull/542. If you have any additional thoughts or insight, I'd recommend that you add them to https://github.com/aws/karpenter-core/issues/370 as we're still actively fleshing out a lot of this design.

**Comment by thjwhite** [comment]
The linked PR has been open for a month, and was just marked stale. if there is no progress on that one, can we re-open this issue or link to another active issue/PR?
